broker/fragment/store_azure.go

```diff
@@ -183,17 +225,48 @@ func (a *azureBackend) azureClient(ep *url.URL) (cfg AzureStoreConfig, client pi
 		return
 	}
 
+	var credentials azblob.Credential
+
 	accountName := os.Getenv("AZURE_ACCOUNT_NAME")
-	accountKey := os.Getenv("AZURE_ACCOUNT_KEY")
-	credentials, err := azblob.NewSharedKeyCredential(accountName, accountKey)
-	if err != nil {
-		return
+	a.endpoint = fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	if ep.Scheme == "azure" {
+		accountKey := os.Getenv("AZURE_ACCOUNT_KEY")
```

Comment:
nit: code style in this repo is to prefer `var accountKey = ...` over `accountKey := ` for new variable initialization.